pkgs/tools/package-management/poetry/default.nix

```
@@ -16,120 +15,28 @@ let
-    pkginfo
-    platformdirs
-    poetry-core
-    poetry-plugin-export
-    requests
-    requests-toolbelt
-    shellingham
-    tomlkit
-    trove-classifiers
-    virtualenv
-  ] ++ lib.optionals (stdenv.isDarwin) [
-    xattr
-  ] ++ lib.optionals (pythonOlder "3.11") [
-    tomli
-  ] ++ lib.optionals (pythonOlder "3.10") [
-    importlib-metadata
-  ] ++ lib.optionals (pythonOlder "3.8") [
-    backports-cached-property
-  ] ++ cachecontrol.optional-dependencies.filecache;
-
-  postInstall = ''
-    installShellCompletion --cmd poetry \
-      --bash <($out/bin/poetry completions bash) \
-      --fish <($out/bin/poetry completions fish) \
-      --zsh <($out/bin/poetry completions zsh) \
-  '';
-
-  # Propagating dependencies leaks them through $PYTHONPATH which causes issues
-  # when used in nix-shell.
-  postFixup = ''
-    rm $out/nix-support/propagated-build-inputs
-  '';
-
-  nativeCheckInputs = with python.pkgs; [
-    cachy
-    deepdiff
-    flatdict
-    pytestCheckHook
-    httpretty
-    pytest-mock
-    pytest-xdist
-  ];
-
-  preCheck = (''
-    export HOME=$TMPDIR
-  '' + lib.optionalString (stdenv.isDarwin && stdenv.isAarch64) ''
-    # https://github.com/python/cpython/issues/74570#issuecomment-1093748531
-    export no_proxy='*';
-  '');
-
-  postCheck = lib.optionalString (stdenv.isDarwin && stdenv.isAarch64) ''
-    unset no_proxy
-  '';
-
-  disabledTests = [
-    # touches network
-    "git"
-    "solver"
-    "load"
-    "vcs"
-    "prereleases_if_they_are_compatible"
-    "test_executor"
-    # requires git history to work correctly
-    "default_with_excluded_data"
-    # toml ordering has changed
-    "lock"
-    # fs permission errors
-    "test_builder_should_execute_build_scripts"
-  ] ++ lib.optionals (python.pythonAtLeast "3.10") [
-    # RuntimeError: 'auto_spec' might be a typo; use unsafe=True if this is intended
-    "test_info_setup_complex_pep517_error"
-  ];
-
-  # Allow for package to use pep420's native namespaces
-  pythonNamespaces = [
-    "poetry"
-  ];
-
-  meta = with lib; {
-    changelog = "https://github.com/python-poetry/poetry/blob/${src.rev}/CHANGELOG.md";
-    homepage = "https://python-poetry.org/";
-    description = "Python dependency management and packaging made easy";
-    license = licenses.mit;
-    maintainers = with maintainers; [ jakewaksbaum dotlambda ];
-  };
-}
+  # pluginSelector is a function mapping pythonPackages to a list of plugins
+  withPlugins = pluginSelector: python.pkgs.toPythonApplication (
+  # e.g. poetry.withPlugins (ps: with ps; [ poetry-plugin-up ])
+    python.pkgs.poetry.overridePythonAttrs (old: {
+      propagatedBuildInputs = old.propagatedBuildInputs ++ pluginSelector python.pkgs;
+
+      # Propagating dependencies leaks them through $PYTHONPATH which causes issues
+      # when used in nix-shell.
+      postFixup = ''
+        rm $out/nix-support/propagated-build-inputs
+      '';
+
+      passthru = {
+        inherit plugins withPlugins;
```

What about adding
```suggestion
        inherit plugins withPlugins withAllPlugins;
```
and adding a corresponding expression to `all-packages.nix`. That's an easy way to ensure the plugins are built by hydra.

---

I added `poetry-with-plugins` to `all-packages.nix`.

---

Decided to with a top-level attribute `poetryPlugins` instead. That works nicely with nixpkgs-review.